lib/tools/adb-commands.js

```diff
@@ -1495,4 +1495,57 @@ methods.bugreport = async function (timeout = 120000) {
   return await this.adbExec(['bugreport'], {timeout});
 };
 
+/**
+ * @typedef {Object} ScreenrecordOptions
+ * @property {?string} videoSize - The format is widthxheight.
+ *                  The default value is the device's native display resolution (if supported),
+ *                  1280x720 if not. For best results,
+ *                  use a size supported by your device's Advanced Video Coding (AVC) encoder.
+ *                  For example, "1280x720"
+ * @property {?boolean} bugReport - Set it to `true` in order to display additional information on the video overlay,
+ *                                  such as a timestamp, that is helpful in videos captured to illustrate bugs.
+ *                                  This option is only supported since API level 27 (Android P).
+ * @property {?string|number} timeLimit - The maximum recording time, in seconds.
+ *                                        The default (and maximum) value is 180 (3 minutes).
+ * @property {?string|number} bitRate - The video bit rate for the video, in megabits per second.
+ *                The default value is 4. You can increase the bit rate to improve video quality,
+ *                but doing so results in larger movie files.
+ */
+
+/**
+ * Initiate screenrecord utility on the device
+ *
+ * @param {string} destination - Full path to the writable media file destination
+ *                               on the device file system.
+ * @param {?ScreenrecordOptions} options [{}]
+ * @returns {SubProcess} screenrecord process, which can be then controlled by the client code
+ */
+methods.screenrecord = async function (destination, options = {}) {
+  const cmd = ['screenrecord'];
+  const {
+    videoSize,
+    bitRate,
+    timeLimit,
+    bugReport,
+  } = options;
+  if (util.hasValue(videoSize)) {
+    cmd.push('--size', videoSize);
+  }
+  if (util.hasValue(timeLimit)) {
+    cmd.push('--time-limit', timeLimit);
+  }
+  if (util.hasValue(bitRate)) {
+    cmd.push('--bit-rate', bitRate);
+  }
+  if (bugReport) {
+    cmd.push('--bugreport');
+  }
+  cmd.push(destination);
+  return new SubProcess(this.executable.path, [
```

Comment:
we're intentionally just returning the process obj and not starting it?

Reply:
yes, we need full control over it. See https://github.com/appium/appium-android-driver/pull/399/files for more details